packages/logrus/challenge-2-structured-logging-and-fields/run_test.sh

```
@@ -0,0 +1,71 @@
+TEMP_DIR=$(mktemp -d)
+
+# Copy the participant's solution, test file, and go.mod/go.sum to the temporary directory
+cp "$SUBMISSION_FILE" "solution-template_test.go" "go.mod" "go.sum" "$TEMP_DIR/" 2>/dev/null
+
+# The test file expects to be in the `main` package alongside the functions it's testing
+mv "$TEMP_DIR/solution.go" "$TEMP_DIR/solution-template.go"
+
+echo "Running tests for user '$USERNAME'..."
+
+# Navigate to the temporary directory
+pushd "$TEMP_DIR" > /dev/null
```

Shellcheck (0.11.0)

[warning] Use 'pushd ... || exit' or 'pushd ... || return' in case pushd fails.

(SC2164)